blocks/bricks/recurrent.py

```diff
@@ -300,6 +301,109 @@ def apply(self, input_=None, state=None, mask=None):
         return next_state
 
 
+class LSTM(BaseRecurrent, Initializable):
+    u"""Long Short Term Memory.
+
+    Every unit of an LSTM is equipped with input, forget and output gates.
+    This implementation aims to do as many computations in parallel as
+    possible and expects the last dimension of the input to be four times
+    the output dimension.
+
+    Parameters
+    ----------
+    dim : int
+        The dimension of the hidden state.
+
+    Notes
+    -----
+    See :class:`.Initializable` for initialization parameters.
```

Comment:
Hi there,
Here's the references:
[1] Gers, Felix A., Nicol N. Schraudolph, and Jürgen Schmidhuber. "Learning precise timing with LSTM recurrent networks." The Journal of Machine Learning Research 3 (2003): 115-143.
[2] Graves, Alex. "Generating sequences with recurrent neural networks." arXiv preprint arXiv:1308.0850 (2013).